contents/docs/runbook/services/plugin-server/ingestion-lag.mdx

```diff
@@ -0,0 +1,211 @@
+
+Also, beyond measuring how fast we are, the ingestion lag metric being high might also be an **indication that we're not processing events at all**, so it's important that we treat it seriously.
+
+## Debugging
+
+There are many things that can lead to ingestion lag being high, so it's very important to rule out parts of the system completely in order to determine the root cause.
+
+The **first thing** to establish is **if we're processing events with a delay** or **not processing events at all**. A third option to consider later on is if we're processing some events and dropping others.
+
+You can do this in many ways:
+
+1. Look at the live events tab
+2. Send an event and see if it appears on live events
+3. Check ClickHouse for the latest event `_timestamp`, and ingestion rate per minute
+
+    ```sql
+    SELECT
+        toStartOfMinute(timestamp) ts,
+        count(*)
+    FROM events
+    WHERE timestamp > now() - INTERVAL 2 HOUR AND timestamp < now()
+    GROUP BY ts
+    ORDER BY ts
+    ```
+
+If we're not processing events, then things are very serious. However, even then there are levels to the issue.
+
+The **worst case scenario** is if events are being dropped by the capture endpoint (`posthog/api/capture.py`). This is because **these events are forever lost**.
+
+It's a smaller but still serious issue if the plugin server is unable to process events or ClickHouse unable to consume them. These events will remain in Kafka until they're correctly processed.
+
+To determine this, there are a few useful resources:
+
+- Graph: Kafka Production Rate ([US Cloud link](https://metrics.posthog.net/d/27BWK5nGk/kafka-msk-overview?orgId=1&viewPanel=22))
+- Graph: Events received per minute ([US Cloud link](https://metrics.posthog.net/d/iffHpmSMz/posthog-cloud?orgId=1&refresh=30s&viewPanel=10))
+- Graph: HTTP Response Codes ([US Cloud link](https://metrics.posthog.net/d/iffHpmSMz/posthog-cloud?orgId=1&refresh=30s&viewPanel=4))
+- Sentry errors for `/e`, `/batch` etc endpoints
+- Plugin server logs
+
+
+## Actions
+
+### Restarting the ingestion pods
+
+A reasonably safe operation to perform is restarting the plugin server ingestion pods. This doesn't cause event loss and can be done by triggering a deploy or running a command like the following:
+
+```shell
+kubectl rollout restart deployment/plugins-ingestion -n posthog
+```
+
+It's ok to do this when you're not fully sure what's wrong with the ingestion pipeline.
+
+### Scaling the ingestion pods
+
+If you suspect that we're struggling due to high load, then you can scale vertically or horizontally.
+
+To scale vertically (i.e. give the pods more resources), you should update the requests and limits for CPU and memory on the [relevant section of the `posthog-cloud-infra` repo](https://github.com/PostHog/posthog-cloud-infra/blob/main/helm/values/common.yml#L137).
```

Comment:
to make this more helpful to link to self-hosted folks - let's call out what values need to be changed in our chart & link to the location of the values file we have in the cloud repo somewhere near the top, where we can have a small section for PostHog specific stuff. For graphs and everything too, ideally those would be ingrained in the chart but we can ignore this part for now.